src/routes/api/sections/querySection.ts

```diff
@@ -1,58 +1,113 @@
 import { Request, Response } from 'express';
 import sectionModel from '../../../models/section.model';
 import timeblockModel from '../../../models/timeblock.model';
+import Room from '../../../models/Room';
+import { DateTime } from 'luxon';
 
 /**
  * Route to query all sections or individual section
  * @author Ross Cleary, Kevin Wang
  */
+
 const querySection = async (req: Request, res: Response): Promise<void> => {
     const { id, startsAt, endsAt } = req.query;
-    if (id) {
-        // If an id is included in the query
-        // Retrieves the room section with the id given in the link
+
+    if (id && startsAt && endsAt) {
+        // If an id and time range is included in the query, retrieves the room section with the id given in the link
         const sectionInformation = await sectionModel.findOne({ _id: id }).catch((error) => {
-            res.status(404).json({ error });
+            res.status(500).json({ error });
             return;
         });
+
         if (!sectionInformation) {
             res.status(404);
             return;
         }
 
-        if (startsAt && endsAt) {
-            const start = new Date(startsAt as string);
-            const end = new Date(endsAt as string);
-            // Validate date strings
-            if (start === null || end === null) {
-                res.status(400).json({ error: 'validation failed, variable types are incorrect' });
+        const start = DateTime.fromISO(String(startsAt), {zone: 'America/Toronto'});
+        const end = DateTime.fromISO(String(endsAt), {zone: 'America/Toronto'});
+        // Validates start and end dates
```

Comment:
We should set a cap for querying times (maybe 31 days should be the maximum allowed?)